### PR TITLE
Add `prefer-node-remove` rule

### LINKED ISSUE #213: Rule proposal: `prefer-node-remove`
Prefer [`node.remove()`](https://developer.mozilla.org/en-US/docs/Web/API/ChildNode/remove) over [`node.parentNode.removeChild(node)`](https://developer.mozilla.org/en-US/docs/Web/API/Node/removeChild). `removeChild` is the legacy way of doing it.

## Fail

```js
foo.parentNode.removeChild(foo);
this.parentNode.removeChild(this);
```

## Pass

```js
foo.remove();
this.remove();
```

### PR DESCRIPTION
**Pull request by medusalix**
Resolves #213

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -54,6 +54,7 @@ module.exports = {
 				'unicorn/no-unreadable-array-destructuring': 'error',
 				'unicorn/no-unused-properties': 'off',
 				'unicorn/prefer-node-append': 'error',
+				'unicorn/prefer-node-remove': 'error',
```
Comment (sindresorhus) [inline index.js]: Should be added to the bottom.

File: docs/rules/prefer-node-remove.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+this.parentNode.removeChild(this);
+```
+
+## Pass
+
+```js
+foo.parentNode.removeChild(bar),
+this.parentNode.removeChild(foo),
+foo.remove(),
+this.remove(),
```
Comment (sindresorhus) [inline docs/rules/prefer-node-remove.md]: Use semicolon

**Comment by medusalix** [comment]
Is there still something you'd want me to change? @sindresorhus

File: test/prefer-node-remove.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+			errors: [{
+				message: 'Prefer `remove` over `removeChild`'
+			}]
+		},
+		{
+			code: 'parentNode.removeChild(this)',
+			output: 'this.remove()',
+			errors: [{
+				message: 'Prefer `remove` over `removeChild`'
+			}]
```
Comment (sindresorhus) [inline test/prefer-node-remove.js]: Can you move the error array to a variable so it doesn't have to be repeated for each case?

File: test/prefer-node-remove.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+ruleTester.run('prefer-node-remove', rule, {
+	valid: [
+		'foo.parentNode.removeChild(bar)',
+		'this.parentNode.removeChild(foo)',
```
Comment (sindresorhus) [inline test/prefer-node-remove.js]: I think we should error on these too (line 13 and 14), but just don't auto-fix.

Comment (sindresorhus) [reply test/prefer-node-remove.js]: Good point. I incorrectly thought that would change the semantics.

Comment (medusalix) [reply test/prefer-node-remove.js]: Oh wow, I didn't even think of it from that point of view. That makes the whole rule a lot simpler 😄.

**Comment by sindresorhus** [comment]
I just discovered the other day in a different project that there's something called [`.parentElement`](https://developer.mozilla.org/en-US/docs/Web/API/Node/parentElement) (not `node`). We should support that too (also needs tests and docs).

File: docs/rules/prefer-node-remove.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# Prefer `remove` over `parentNode.removeChild` or `parentElement.removeChild`
```
Comment (sindresorhus) [inline docs/rules/prefer-node-remove.md]:
```suggestion
# Prefer `remove` over `parentNode.removeChild` and `parentElement.removeChild`
```

File: rules/prefer-node-remove.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+	if (identifier.type === 'ThisExpression') {
+		return 'this';
+	}
+
+	if (identifier.type === 'Identifier') {
+		return identifier.name;
+	}
+
+	return null;
+};
```
Comment (sindresorhus) [inline rules/prefer-node-remove.js]: Are any of these methods maybe generally useful and generic enough to be moved to a `util.js` file?

Comment (medusalix) [reply rules/prefer-node-remove.js]: Yes, I think `getMethodName` and `getCallerName` would generally be useful. `getArgumentName` only works for a single argument to a function, so I don't think that one would be used much.

**Comment by futpib** [comment]
@medusalix ignoring `foo.parentNode['removeChild'](foo);` is okay.

File: rules/prefer-node-remove.js (hunk 6 above)
Comment (sindresorhus) [reply rules/prefer-node-remove.js]: Would be cool if you could extract them in a separate PR.

**Comment by sindresorhus** [comment]
Nice work, @medusalix 👍

**Pull request merged**